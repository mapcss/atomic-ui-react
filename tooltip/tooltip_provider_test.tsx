import TooltipProvider from "./tooltip_provider.ts";
import Tooltip from "./tooltip.ts";
import {
  assertSnapshot,
  describe,
  expect,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import { renderToStaticMarkup } from "react-dom/server";
import { fireEvent, render } from "@testing-library/react";

describe("TooltipProvider", () => {
  it("should render markup", (t) => {
    assertSnapshot(
      t,
      renderToStaticMarkup(<TooltipProvider>{() => <></>}</TooltipProvider>),
    );
    assertSnapshot(
      t,
      renderToStaticMarkup(
        <TooltipProvider as="span">
          {() => <></>}
        </TooltipProvider>,
      ),
    );
  });
});

const describeTooltipProvider = describe({
  name: "TooltipProvider",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(
  describeTooltipProvider,
  "should render tooltip when mouseenter, remove tooltip when mouseleave",
  (t) => {
    const result = render(
      <TooltipProvider>
        {({ isShow, ref }) => (
          <>
            <button data-testid="test" ref={ref}></button>
            {isShow && <Tooltip>Tooltip</Tooltip>}
          </>
        )}
      </TooltipProvider>,
    );

    assertSnapshot(t, result.baseElement.innerHTML);
    const target = result.getByTestId("test");
    fireEvent.mouseEnter(target);
    assertSnapshot(t, result.baseElement.innerHTML);
    fireEvent.mouseLeave(target);
    assertSnapshot(t, result.baseElement.innerHTML);
  },
);
