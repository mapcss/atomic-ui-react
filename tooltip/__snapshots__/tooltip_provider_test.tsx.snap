export const snapshot = {};

snapshot[`TooltipProvider > should render markup 1`] = `'<div style="position:relative"></div>'`;

snapshot[`TooltipProvider > should render markup 2`] = `'<span style="position:relative"></span>'`;

snapshot[`TooltipProvider > should render tooltip when mouseenter, remove tooltip when mouseleave 1`] = `'<div><div style="position: relative;"><button data-testid="test"></button></div></div>'`;

snapshot[`TooltipProvider > should render tooltip when mouseenter, remove tooltip when mouseleave 2`] = `'<div><div style="position: relative;"><button data-testid="test"></button><span role="tooltip">Tooltip</span></div></div>'`;

snapshot[`TooltipProvider > should render tooltip when mouseenter, remove tooltip when mouseleave 3`] = `'<div><div style="position: relative;"><button data-testid="test"></button></div></div>'`;
